### PR TITLE
[DOCS] Relocate search API's request body parameters

### PR DESCRIPTION
**Pull request by jrodewig**
Changes:
* Moves the request body parameters for the search API
  from the Request body search page to the Search API reference page.

* Relocates a search request body example from the Request body search
  page to the Search API reference page.

* Adds a note to any duplicated query and request body parameters.

### CODE REVIEW

File: docs/reference/search/request-body.asciidoc [hunk 1]
```diff
@@ -37,106 +36,10 @@ The search request can be executed with a search DSL, which includes the
...
-  (Optional, integer) The number of shard results that should be reduced at once 
-  on the coordinating node. This value should be used as a protection mechanism 
-  to reduce the memory overhead per search request if the potential number of 
-  shards in the request can be large.
-
-[[ccs-minimize-roundtrips]]
-`ccs_minimize_roundtrips`::
-  (Optional, boolean) If `true`, the network round-trips between the 
-  coordinating node and the remote clusters ewill be minimized when executing 
-  {ccs} requests. See <<ccs-network-delays>>. Defaults to `true`.
```
Comment (cbuescher) [inline docs/reference/search/request-body.asciidoc:61]: On the pages this is moved to, cross-cluster search ({ccs}) as a term isn't mentioned explicitly and the link to the network delays is missing. I don't know if this was left out on purpose to simplify things, but at least spelling out the abreviation might be a good addition.

File: docs/reference/search/request-body.asciidoc [hunk 2]
```diff
@@ -37,106 +36,10 @@ The search request can be executed with a search DSL, which includes the
...
-`ccs_minimize_roundtrips`::
-  (Optional, boolean) If `true`, the network round-trips between the 
-  coordinating node and the remote clusters ewill be minimized when executing 
-  {ccs} requests. See <<ccs-network-delays>>. Defaults to `true`. 
-
-include::{docdir}/rest-api/common-parms.asciidoc[tag=from]
-
-`request_cache`::
-  (Optional, boolean) If `true`, the caching of search results is enabled for 
-  requests where `size` is `0`. See <<shard-request-cache>>.
```
Comment (cbuescher) [inline docs/reference/search/request-body.asciidoc:67]: > is enabled for requests where `size` is `0`
I don't know if this is useful information, seems to be missing from the target page

File: docs/reference/search/request-body.asciidoc [hunk 3]
```diff
@@ -37,106 +36,10 @@ The search request can be executed with a search DSL, which includes the
...
-`size`:: 
-  (Optional, integer) The number of hits to return. Defaults to `10`.
-
-include::{docdir}/rest-api/common-parms.asciidoc[tag=terminate_after]
-
-include::{docdir}/rest-api/common-parms.asciidoc[tag=search_timeout]
-
-
-Out of the above, the `search_type`, `request_cache` and the 
-`allow_partial_search_results` settings must be passed as query-string
```
Comment (cbuescher) [inline docs/reference/search/request-body.asciidoc:80]: does this in reverse mean the `batched_reduce_size` and `ccs_minimize_roundtrips` should also go to the "body" section on the target page? Currently they seem to be only listed under Query Parameters. I don't know if they are allowed in both places tbh but we can check...

**Review by cbuescher — COMMENTED** [review]
I like the reorganization, only left a couple of comments where some information might have been dropped on the move and a few questions around parameters that should maybe go to the "body" section rather than the "parameters". The list of whats allowed in parameters vs. body might need to be rechecked at some point, but that can probably be done in a follow up.

File: docs/reference/search/request-body.asciidoc (hunk 3 above)
Comment (jrodewig) [reply docs/reference/search/request-body.asciidoc:80]: I tested this by running off the `master` branch and it appears that neither `batched_reduce_size` nor `ccs_minimize_roundtrips` are valid request body parameters. Both work fine as query parameters though.

### Setup
Index a simple 
```
PUT /my_index/_doc/0
{
  "user": "kimchy"
}
```

### `ccs_minimize_roundtrips`
```
GET /my_index/_search
{
    "ccs_minimize_roundtrips": false,
    "query" : {
        "term" : { "user" : "kimchy" }
    }
}
```
Response:

```
{
  "error" : {
    "root_cause" : [
      {
        "type" : "parsing_exception",
        "reason" : "Unknown key for a VALUE_BOOLEAN in [ccs_minimize_roundtrips].",
        "line" : 3,
        "col" : 32
      }
    ],
    "type" : "parsing_exception",
    "reason" : "Unknown key for a VALUE_BOOLEAN in [ccs_minimize_roundtrips].",
    "line" : 3,
    "col" : 32
  },
  "status" : 400
}
```

### `batched_reduce_size`
```
GET /my_index/_search
{
    "batched_reduce_size": 512,
    "query" : {
        "term" : { "user" : "kimchy" }
    }
}
```

Response:

```
{
  "error" : {
    "root_cause" : [
      {
        "type" : "parsing_exception",
        "reason" : "Unknown key for a VALUE_NUMBER in [batched_reduce_size].",
        "line" : 3,
        "col" : 28
      }
    ],
    "type" : "parsing_exception",
    "reason" : "Unknown key for a VALUE_NUMBER in [batched_reduce_size].",
    "line" : 3,
    "col" : 28
  },
  "status" : 400
}
```

File: docs/reference/search/request-body.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/search/request-body.asciidoc:61]: Thanks for catching this. I agree this is the better definition and moved it over with 013cca62d65. As a note: `{ccs}` expands to `cross-cluster search`, not `CCS`, so the full wording renders in the final HTML.

File: docs/reference/search/request-body.asciidoc (hunk 2 above)
Comment (jrodewig) [reply docs/reference/search/request-body.asciidoc:67]: Another good catch. Moved over with 013cca62d65.

**Comment by jrodewig** [comment]
Thanks for taking a look @cbuescher. I made some adjustments based on your feedback.

File: docs/reference/search/request-body.asciidoc (hunk 3 above)
Comment (cbuescher) [reply docs/reference/search/request-body.asciidoc:80]: Thanks for checking.

**Review by cbuescher — APPROVED** [review]
Thanks for the update, LGTM

**Comment by jrodewig** [comment]
Thanks @cbuescher

**Comment by jrodewig** [comment]
### Backport commits
master b5f219c
7.x ea76b0c 
7.8 276e86c
7.7 0f42122 
7.6 bb381ea

**Pull request merged**